selftests/unit/test_utils_service.py

```diff
@@ -26,6 +26,30 @@
 from avocado.utils import service
 
 
+class TestRunCalls(unittest.TestCase):
+
+    def setUp(self):
+        self.run_param1 = ["foo_target", "set_target", mock.Mock()]
+        self.run_param2 = ["foo_service", "start", mock.Mock()]
+        self.run_param1[-1].return_value.stdout = "systemd"
+        self.run_param2[-1].return_value.stdout = "init"
+        self.results = ["systemctl isolate foo_target",
+                        "service foo_service start"]
```

Comment:
Regarding this selftes, how many times do you expect this setup to be used? I mean calling `set_target` followed by `start` on 2 different instances... IMO this is way too specific to be a setup. The test itself is actually more generic than this setup, which is quite odd and makes it hard to follow. Unittests are meant to be simple, how about something like:


```python
class TestMultipleInstances(unittest.TestCase):

    def test_different_runners(self):
        # Call 'set_target' on first runner
        runner1 = mock.Mock()
        runner1.return_value.stdout = 'systemd'
        service1 = service.service_manager(run=runner1)
        service1.set_target('foo_target')
        self.assertEqual(runner1.call_args[0][0],
                         'systemctl isolate foo_target')
        # Call 'start' on second runner
        runner2 = mock.Mock()
        runner2.return_value.stdout = 'init'
        service2 = service.service_manager(run=runner2)
        service2.start('foo_service')
        self.assertEqual(runner2.call_args[0][0],
                         'service foo_service start')

```

I think it'd be easier to follow and see what is happening (yes, it's not that nice and it repeats patterns, but the readability is IMO way better, what do you think?)